bip-taro-proof-file.mediawiki

```diff
@@ -223,6 +238,19 @@ verify_asset_file_proof(file_proof []byte, genesis_outpoint OutPoint,
             ):
                 return false
 
+        has_meta_reveal = proof_tlv_map.meta_reveal is not None
+        is_genesis_asset = is_genesis_asset(proof_tlv_map.asset)
+        match:
+            case has_meta_reveal && !is_genesis_asset:
+                return false
+
+            case has_meta_reveal && is_genesis_asset:
+                meta_hash := sha256(meta_reveal)
+                if meta_hash != proof_tlv_map.asset.meta_hash:
+                    return false
+            case !has_meta_reveal && is_genesis_asset:
```

Comment:
Pseudocode requires a meta reveal but `bip-taro` says the preimage is optional?

Reply:
Revised it, check now?